mtp_api/apps/security/tests/test_views.py

```diff
@@ -1076,18 +1078,63 @@ def test_check_filtering_by_credit_resolution(self):
             Check.objects.filter(status=CHECK_STATUS.PENDING).count() - 1,
         )
 
+    def test_check_filtering_by_actioned_by(self):
+        """
+        Test that the list endpoint only returns the checks with an actioned_by id.
+        """
+        filters = {
+            'actioned_by': True,
+        }
+
+        auth = self.get_http_authorization_for_user(self._get_authorised_user())
+        response = self.client.get(
+            reverse('security-check-list'),
+            filters,
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+
+        self.assertEqual(response.status_code, http_status.HTTP_200_OK)
+
+        response_data = response.json()
+
+        self.assertEqual(
+            response_data['count'],
```

Comment:
instead of just checking you have the right number, i think you should look at the checks returned and see that they were filtered as expected

Reply:
I feel this test does this for us anyway? As we are asserting that we only want a response with checks which have an `'actioned_by': True` which means only checks which have an actioned_by id. Then we are checking the DB to only grab the checks which have an `actioned_by` so when asserting this should be enough? Because the query to the DB will only bring back the checks which have an actioned_by which is not null. I tried to keep the tests similar to the ones above for consistency purposes also.

Reply:
From this test I feel this gives me enough confidence it is filtering the way we expect

Reply:
>Then we are checking the DB to only grab the checks which have an actioned_by
you are not. you're only checking that it's the same _number_

Reply:
the point of this test is to _ensure_ that the filter `{'actioned_by': true}` does the right thing, which is returning only non-nulls. this test only proves you get the right number back. but what you want is for it to make sure the contract is maintained if any further changes happen on the view

Reply:
Okay this makes more sense. Ill implement this